src/renderer/actions/paths.js

```diff
@@ -81,24 +174,28 @@ const validateMac = function(channel, proposedPath) {
     return "";
 };
 
+
 const validateLinux = function(channel, proposedPath) {
-    if (proposedPath.includes("/snap/")) {
-        remote.dialog.showErrorBox("BetterDiscord Incompatible", "BetterDiscord is currently incompatible with Snap installs of Discord. Support for snap installs is coming soon!");
-        return "";
-    }
+    // Regardless of whether it's a Flatpak, Snap or Native, we want to search
+    // for their "inner xdg-config/discord[channel]" folder which are all name
+    // the same regardless of installation method, so we treat the channel name
+    // detection identically for all proposed paths.
     const channelName = platforms[channel].toLowerCase().replace(" ", "");
 
+    // Attempt to detect which folder the user has selected.
     let resourcePath = "";
     const selected = path.basename(proposedPath);
-    if (selected === channelName) {
-        const version = fs.readdirSync(proposedPath).filter(f => fs.lstatSync(path.join(proposedPath, f)).isDirectory() && f.split(".").length > 1).sort().reverse()[0];
+    if (selected === channelName) {  // "discord", "discordcanary", etc.
+        const version = findLatestDiscordVersion(proposedPath);
         if (!version) return "";
         resourcePath = path.join(proposedPath, version, "modules", "discord_desktop_core");
     }
-    if (selected.split(".").length > 2) resourcePath = path.join(proposedPath, "modules", "discord_desktop_core");
-    if (selected === "modules") resourcePath = path.join(proposedPath, "discord_desktop_core");
-    if (selected === "discord_desktop_core") resourcePath = proposedPath;
+    else if (selected.split(".").length > 1) resourcePath = path.join(proposedPath, "modules", "discord_desktop_core");
```

Comment:
Inconsistent bracket use in combination with else-if

Reply:
You want me to add explicit `{}` brackets in the if-elseif-else chain here? I just omitted the brackets since it seemed your current code-style preferred no brackets for single-command statements.